.github/workflows/android.yaml

```
@@ -0,0 +1,68 @@
+name: Android Device Tests
+
+on:
+  push:
```

```suggestion
on:
  push:
    branches: [ main ]
  pull_request:
```

let's align to `ci.yaml`

---

sgtm. i'll just leave this comment open until we are happy with the PR so it keeps running on my fork :) and will fix it before merge.